teps/0056-pipelines-in-pipelines.md

```diff
@@ -2,283 +2,389 @@
+                                   git-clone
+                                       |
+                                       v
+                                     build
+                                       |
+                                       v
+           ----------------------------------------------------------
+           |              |            |             |              |
+           v              v            v             v              v          
+    apply-configmap   apply-pvc   apply-route   apply-service     apply-dc
+    ----------------------------------------------------------------------
+               |                   |                       |
+               v                   v                       v   
+          add-configmap    add-columna-service       add-kafka-config      
+          ------------------------------------------------------------
+                                   |                           
+                                   v                           
+                                deploy
 ```
 
-When editing TEPS, aim for tightly-scoped, single-topic PRs to keep discussions
-focused.  If you disagree with what is already in a document, open a new PR
-with suggested changes.
+Instead, I want to define and distribute `apply-config` set of 5 `Tasks` and `add-config` set of 3 `Tasks` so that I
+can specify that the latter waits for the former to complete execution.
 
-If there are new details that belong in the TEP, edit the TEP.  Once a
-feature has become "implemented", major changes should get new TEPs.
+```                           
+       git-clone
+           |                           
+           v                           
+         build 
+           |
+           v 
+      apply-config
+           |                           
+           v                           
+       add-config 
+           |
+           v 
+        deploy
+```
 
-The canonical place for the latest set of instructions (and the likely source
-of this file) is [here](/teps/NNNN-TEP-template/README.md).
+For further information, read [issue][issue-2134] and [issue][issue-4067].
 
--->
+### Reusability
 
-<!--
-This is the title of your TEP.  Keep it short, simple, and descriptive.  A good
-title can help communicate what the TEP is and should be considered as part of
-any review.
--->
+Today, a subset of `Tasks` in a `Pipeline` cannot be distributed as a unit of execution within `Pipelines`. As 
+such, users have to specify those related `Tasks` separately and repeatedly across many `Pipelines`.
 
-<!--
-A table of contents is helpful for quickly jumping to sections of a TEP and for
-highlighting any additional information provided beyond the standard TEP
-template.
+As a `Pipeline` author, I need to define a set of `Tasks` as a complete unit of execution that I can share across
+`Pipelines`. For example, I can define and distribute `linting` set of `Tasks` and `testing` set of `Tasks`:
```

Comment:
I will argue with this particular use case 🙃 

I think as a community if possible, we should advise our users to transform such tasks into a list of steps and make them part of a single task. But at the same time, it depends on how these tasks are configured and executed. The `tasks` being the only sharable unit of execution and a single point of failure, we have to rely on splitting such process into multiple tasks. One alternative here is to offer guidance such that `linting` is a step with `onError` configured to `continue` such that `report-linter-output` is executed regardless.

Reply:
makes sense, updated the use cases to the user-provided example in https://github.com/tektoncd/pipeline/issues/2134#issuecomment-631552148, thank you!

Reply:
thank you @jerop 👍